build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -22,24 +22,77 @@ crate::define_env_var! {
     GRAALVM_HOME, PathBuf;
 }
 
-pub fn graal_version_from_version_string(version_string: &str) -> Result<Version> {
-    let line = version_string.lines().find(|line| line.contains("GraalVM CE")).context(
+const CE_JAVA_VENDOR: &str = "GraalVM CE";
+const EE_JAVA_VENDOR: &str = "Oracle GraalVM";
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Edition {
+    /// GraalVM CE (Community Edition).
+    Community,
+    /// Oracle GraalVM (Formerly GraalVM EE, Enterprise Edition)
+    Enterprise,
+}
+
+impl std::str::FromStr for Edition {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s {
+            CE_JAVA_VENDOR => Ok(Edition::Community),
+            EE_JAVA_VENDOR => Ok(Edition::Enterprise),
+            _ => bail!("Unknown GraalVM edition: {}", s),
+        }
+    }
+
+}
+
+impl Into<String> for Edition {
+    fn into(self) -> String {
+        match self {
+            Edition::Community => CE_JAVA_VENDOR.to_string(),
+            Edition::Enterprise => EE_JAVA_VENDOR.to_string(),
+        }
+    }
+}
+
+impl Display for Edition {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        match *self {
+            Edition::Community => write!(f, "{}", CE_JAVA_VENDOR),
+            Edition::Enterprise => write!(f, "{}", EE_JAVA_VENDOR),
+        }
+    }
+}
+
+
+fn graal_version_from_version_string(version_string: &str) -> Result<(Version, Edition)> {
+    let line = version_string.lines().find(|line| line.contains(CE_JAVA_VENDOR) || line.contains(EE_JAVA_VENDOR)).context(
         "There is a Java environment available but it is not recognizable as GraalVM one.",
     )?;
-    Version::find_in_text(line)
+    let edition = if line.contains(CE_JAVA_VENDOR) {
+        Edition::Community
+    } else if line.contains(EE_JAVA_VENDOR) {
+        Edition::Enterprise
+    } else {
+        bail!("Unknown GraalVM edition")
+    };
+    let version = Version::find_in_text(line);
+    version.map(|version| (version, edition)).context("Failed to find GraalVM version")
 }
 
-pub async fn find_graal_version() -> Result<Version> {
+async fn find_graal_version() -> Result<(Version, Edition)> {
```

Comment:
Why this change? We use `pub` by default.

Reply:
I have JVM background. I like to keep things as hidden as possible. This function was not used outside this module, so I simply removed the `pub` modifier. But if declaring everything public is the correct Rust way, I will revert this change.

Reply:
Redeclared as public in 859483a83be87d60ce4c4542cfdaaa27e15cd3b5